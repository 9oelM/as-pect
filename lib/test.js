"use strict";
/**
 * This file is used by the package.json "asp" script to simulate using the command line interface.
 */
require("../lib/cli/index.js").asp(process.argv.slice(2));
//# sourceMappingURL=data:application/json;base64,eyJ2ZXJzaW9uIjozLCJmaWxlIjoidGVzdC5qcyIsInNvdXJjZVJvb3QiOiIiLCJzb3VyY2VzIjpbIi4uL3NyYy90ZXN0LnRzIl0sIm5hbWVzIjpbXSwibWFwcGluZ3MiOiI7QUFBQTs7R0FFRztBQUNILE9BQU8sQ0FBQyxxQkFBcUIsQ0FBQyxDQUFDLEdBQUcsQ0FBQyxPQUFPLENBQUMsSUFBSSxDQUFDLEtBQUssQ0FBQyxDQUFDLENBQUMsQ0FBQyxDQUFDIn0=