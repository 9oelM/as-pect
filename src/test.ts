import { asp } from "./cli";

asp(process.argv.slice(2));
