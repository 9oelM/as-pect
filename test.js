const glob = require("glob");
const path = require("path");
const pattern = path.join(process.cwd(), "assembly/tests/**/*.ts");
console.log(pattern);

console.log(glob.sync(pattern));