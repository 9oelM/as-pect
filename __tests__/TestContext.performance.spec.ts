import { TestContext } from "../src/test/TestContext";
import { createPerformanceModule } from "./setup/createPerformanceModule";

let ctx: TestContext;

let start = new Promise<void>((resolve, reject) => {
  createPerformanceModule({}, (err, result) => {
    if (err) {
      console.log(err);
      reject(err);
    } else {
      ctx = result!;
      resolve();
    }
  });
});

beforeAll(() => start);

describe("pass-fail", () => {
  test("pass-fail", () => {
    for (const group of ctx.testGroups) {
      for (const test of group.tests) {
        expect(test.performance).toMatchSnapshot(`${group.name}~${test.name} performance`);
        expect(test.hasAverage).toMatchSnapshot(`${group.name}~${test.name} hasAverage`);
        expect(test.hasMedian).toMatchSnapshot(`${group.name}~${test.name} hasMedian`);
        expect(test.hasMin).toMatchSnapshot(`${group.name}~${test.name} hasMin`);
        expect(test.hasMax).toMatchSnapshot(`${group.name}~${test.name} hasMax`);
        expect(test.hasStdDev).toMatchSnapshot(`${group.name}~${test.name} hasStdDev`);
        expect(test.hasVariance).toMatchSnapshot(`${group.name}~${test.name} hasVariance`);
      }
    }
  });
});
