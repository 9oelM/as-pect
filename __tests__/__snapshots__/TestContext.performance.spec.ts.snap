// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pass-fail pass-fail: performance with failures in afterEach~should fail hasAverage 1`] = `false`;

exports[`pass-fail pass-fail: performance with failures in afterEach~should fail hasMax 1`] = `false`;

exports[`pass-fail pass-fail: performance with failures in afterEach~should fail hasMedian 1`] = `false`;

exports[`pass-fail pass-fail: performance with failures in afterEach~should fail hasMin 1`] = `false`;

exports[`pass-fail pass-fail: performance with failures in afterEach~should fail hasStdDev 1`] = `false`;

exports[`pass-fail pass-fail: performance with failures in afterEach~should fail hasVariance 1`] = `false`;

exports[`pass-fail pass-fail: performance with failures in afterEach~should fail performance 1`] = `true`;

exports[`pass-fail pass-fail: performance with failures in beforeEach~should fail hasAverage 1`] = `false`;

exports[`pass-fail pass-fail: performance with failures in beforeEach~should fail hasMax 1`] = `false`;

exports[`pass-fail pass-fail: performance with failures in beforeEach~should fail hasMedian 1`] = `false`;

exports[`pass-fail pass-fail: performance with failures in beforeEach~should fail hasMin 1`] = `false`;

exports[`pass-fail pass-fail: performance with failures in beforeEach~should fail hasStdDev 1`] = `false`;

exports[`pass-fail pass-fail: performance with failures in beforeEach~should fail hasVariance 1`] = `false`;

exports[`pass-fail pass-fail: performance with failures in beforeEach~should fail performance 1`] = `true`;

exports[`pass-fail pass-fail: performance~this should have average enabled hasAverage 1`] = `true`;

exports[`pass-fail pass-fail: performance~this should have average enabled hasMax 1`] = `false`;

exports[`pass-fail pass-fail: performance~this should have average enabled hasMedian 1`] = `false`;

exports[`pass-fail pass-fail: performance~this should have average enabled hasMin 1`] = `false`;

exports[`pass-fail pass-fail: performance~this should have average enabled hasStdDev 1`] = `false`;

exports[`pass-fail pass-fail: performance~this should have average enabled hasVariance 1`] = `false`;

exports[`pass-fail pass-fail: performance~this should have average enabled performance 1`] = `true`;

exports[`pass-fail pass-fail: performance~this should have max enabled hasAverage 1`] = `false`;

exports[`pass-fail pass-fail: performance~this should have max enabled hasMax 1`] = `true`;

exports[`pass-fail pass-fail: performance~this should have max enabled hasMedian 1`] = `false`;

exports[`pass-fail pass-fail: performance~this should have max enabled hasMin 1`] = `false`;

exports[`pass-fail pass-fail: performance~this should have max enabled hasStdDev 1`] = `false`;

exports[`pass-fail pass-fail: performance~this should have max enabled hasVariance 1`] = `false`;

exports[`pass-fail pass-fail: performance~this should have max enabled performance 1`] = `true`;

exports[`pass-fail pass-fail: performance~this should have min enabled hasAverage 1`] = `false`;

exports[`pass-fail pass-fail: performance~this should have min enabled hasMax 1`] = `false`;

exports[`pass-fail pass-fail: performance~this should have min enabled hasMedian 1`] = `false`;

exports[`pass-fail pass-fail: performance~this should have min enabled hasMin 1`] = `true`;

exports[`pass-fail pass-fail: performance~this should have min enabled hasStdDev 1`] = `false`;

exports[`pass-fail pass-fail: performance~this should have min enabled hasVariance 1`] = `false`;

exports[`pass-fail pass-fail: performance~this should have min enabled performance 1`] = `true`;

exports[`pass-fail pass-fail: performance~this should have performance enabled hasAverage 1`] = `false`;

exports[`pass-fail pass-fail: performance~this should have performance enabled hasMax 1`] = `false`;

exports[`pass-fail pass-fail: performance~this should have performance enabled hasMedian 1`] = `false`;

exports[`pass-fail pass-fail: performance~this should have performance enabled hasMin 1`] = `false`;

exports[`pass-fail pass-fail: performance~this should have performance enabled hasStdDev 1`] = `false`;

exports[`pass-fail pass-fail: performance~this should have performance enabled hasVariance 1`] = `false`;

exports[`pass-fail pass-fail: performance~this should have performance enabled performance 1`] = `true`;

exports[`pass-fail pass-fail: performance~this should have set the decimal places hasAverage 1`] = `false`;

exports[`pass-fail pass-fail: performance~this should have set the decimal places hasMax 1`] = `false`;

exports[`pass-fail pass-fail: performance~this should have set the decimal places hasMedian 1`] = `false`;

exports[`pass-fail pass-fail: performance~this should have set the decimal places hasMin 1`] = `false`;

exports[`pass-fail pass-fail: performance~this should have set the decimal places hasStdDev 1`] = `false`;

exports[`pass-fail pass-fail: performance~this should have set the decimal places hasVariance 1`] = `false`;

exports[`pass-fail pass-fail: performance~this should have set the decimal places performance 1`] = `true`;

exports[`pass-fail pass-fail: performance~this should have stdDev enabled hasAverage 1`] = `false`;

exports[`pass-fail pass-fail: performance~this should have stdDev enabled hasAverage 2`] = `false`;

exports[`pass-fail pass-fail: performance~this should have stdDev enabled hasMax 1`] = `false`;

exports[`pass-fail pass-fail: performance~this should have stdDev enabled hasMax 2`] = `false`;

exports[`pass-fail pass-fail: performance~this should have stdDev enabled hasMedian 1`] = `false`;

exports[`pass-fail pass-fail: performance~this should have stdDev enabled hasMedian 2`] = `true`;

exports[`pass-fail pass-fail: performance~this should have stdDev enabled hasMin 1`] = `false`;

exports[`pass-fail pass-fail: performance~this should have stdDev enabled hasMin 2`] = `false`;

exports[`pass-fail pass-fail: performance~this should have stdDev enabled hasStdDev 1`] = `true`;

exports[`pass-fail pass-fail: performance~this should have stdDev enabled hasStdDev 2`] = `false`;

exports[`pass-fail pass-fail: performance~this should have stdDev enabled hasVariance 1`] = `true`;

exports[`pass-fail pass-fail: performance~this should have stdDev enabled hasVariance 2`] = `false`;

exports[`pass-fail pass-fail: performance~this should have stdDev enabled performance 1`] = `true`;

exports[`pass-fail pass-fail: performance~this should have stdDev enabled performance 2`] = `true`;

exports[`pass-fail pass-fail: performance~this should have variance enabled hasAverage 1`] = `false`;

exports[`pass-fail pass-fail: performance~this should have variance enabled hasMax 1`] = `false`;

exports[`pass-fail pass-fail: performance~this should have variance enabled hasMedian 1`] = `false`;

exports[`pass-fail pass-fail: performance~this should have variance enabled hasMin 1`] = `false`;

exports[`pass-fail pass-fail: performance~this should have variance enabled hasStdDev 1`] = `false`;

exports[`pass-fail pass-fail: performance~this should have variance enabled hasVariance 1`] = `true`;

exports[`pass-fail pass-fail: performance~this should have variance enabled performance 1`] = `true`;

exports[`pass-fail pass-fail: performance~this should only run for less than a second hasAverage 1`] = `false`;

exports[`pass-fail pass-fail: performance~this should only run for less than a second hasAverage 2`] = `false`;

exports[`pass-fail pass-fail: performance~this should only run for less than a second hasMax 1`] = `false`;

exports[`pass-fail pass-fail: performance~this should only run for less than a second hasMax 2`] = `false`;

exports[`pass-fail pass-fail: performance~this should only run for less than a second hasMedian 1`] = `false`;

exports[`pass-fail pass-fail: performance~this should only run for less than a second hasMedian 2`] = `false`;

exports[`pass-fail pass-fail: performance~this should only run for less than a second hasMin 1`] = `false`;

exports[`pass-fail pass-fail: performance~this should only run for less than a second hasMin 2`] = `false`;

exports[`pass-fail pass-fail: performance~this should only run for less than a second hasStdDev 1`] = `false`;

exports[`pass-fail pass-fail: performance~this should only run for less than a second hasStdDev 2`] = `false`;

exports[`pass-fail pass-fail: performance~this should only run for less than a second hasVariance 1`] = `false`;

exports[`pass-fail pass-fail: performance~this should only run for less than a second hasVariance 2`] = `false`;

exports[`pass-fail pass-fail: performance~this should only run for less than a second performance 1`] = `true`;

exports[`pass-fail pass-fail: performance~this should only run for less than a second performance 2`] = `true`;
